### ISSUE TITLE
Special Handling of Llama Tokenizer Padding

### ISSUE THREAD

**Issue by Joe-Hall-Lee** [YangRui2015/Generalizable-Reward-Model#1]
Hi,
Thank you for your excellent work. I have a question regarding the tokenizer handling in the [reward model training script](https://github.com/YangRui2015/Generalizable-Reward-Model/blob/06f30be01d6281de037371266f94f548e7f223bb/reward_models/run_reward_models_train.py#L101). Specifically, I am curious about the following code snippet:
```
if tokenizer.pad_token == None:
    if 'Llama' in script_args.base_model:
        tokenizer.add_special_tokens({'pad_token': '[PAD]'})
    else:
        tokenizer.pad_token = tokenizer.eos_token
```
I don't quite understand why Llama requires special handling. When I removed the condition and trained with Llama-3.2-3B-Instruct, I observed that the loss remained unchanged. Despite looking it up, I still feel confused.
Could you please explain the rationale behind the special treatment for Llama? Thanks a lot!

**Comment by YangRui2015** [comment]
Yes, I think it is due to the chat template of Llama which includes the eos_token before the response. Since the chosen and rejected share the same prompt, the loss would be unchanged -log sigmoid (0) ～= 0.693. Adding a different pad token for Llama is important to avoid this issue.

**Comment by Joe-Hall-Lee** [comment]
I understand. Thank you.
